### ISSUE TITLE
Test all compilers in Travis which are allowed in the makefile

### ISSUE THREAD

**Issue by LorenzMeier** [PX4/Firmware#2294]
Currently we're not testing all supported compilers in Travis. Current issue to be addressed:

```
PX4/Firmware/src/lib/uavcan/libuavcan/include/uavcan/marshal/array.hpp:516:13: error: comparison is always false due to limited range of data type [-Werror=type-limits]
```

**Comment by LorenzMeier** [comment]
@pavel-kirienko @davids5 It would be great if you could have a look at the error message above.

**Comment by pavel-kirienko** [comment]
What compiler produced this error?

**Comment by kd0aij** [comment]
arm-none-eabi-gcc (GNU Tools for ARM Embedded Processors) 4.7.4 20140401 (release) [ARM/embedded-4_7-branch revision 209195]

**Comment by pavel-kirienko** [comment]
Fixed in master.

**Comment by dagar** [comment]
Do you want to support gcc 4.7-4.9? We could easily split this into 3 travis-ci linux builds that run in parallel, but we'd need to decide which one deploys to s3 and github releases.

**Comment by LorenzMeier** [comment]
4.8 and 4.9 would be good.

**Comment by LorenzMeier** [comment]
Doing that today, thanks @dagar
